### ISSUE TITLE
ES6 version?

### ISSUE THREAD

**Issue by Qualinost1** [JimmyLaurent/torrent-search-api#50]
Hi,
First many thanks for this module which is clearly the best I tried so far for torrents.
I'm just an hobbist and I probably don't understand half of what I'm using so sorry in advance if my question is stupid.
I would love to use your module in an Angular/Electron app but I'm facing the problem of CommonJS module in an ES6 project (require/export ...). I tried many things (like the [cjs-to-es6 module](https://github.com/nolanlawson/cjs-to-es6)) but it's clearly not straightforward and it generates me lots of compilation errors.
Do you have any ambitions to rewrite your module in ES6 ? Or maybe any idea of how can I use your module in an Electron/Angular 6 project?
Oh, and sorry in advance for my english... :)

**Comment by JimmyLaurent** [comment]
I don't plan to use ES6 modules since it's not supported in the stable version of nodejs.

But I don't think your you problem is here, it may be some troubles with the dynamic imports ([here](https://github.com/JimmyLaurent/torrent-search-api/blob/master/lib/ProviderManager.js#L176)) and your transpilation step.

**Comment by Qualinost1** [comment]
> I don't plan to use ES6 modules since it's not supported in the stable version of nodejs.
> 
> But I don't think your you problem is here, it may be some troubles with the dynamic imports ([here](https://github.com/JimmyLaurent/torrent-search-api/blob/master/lib/ProviderManager.js#L176)) and your transpilation step.

@JimmyLaurent : Thank you, you pointed out one of my problems! I had to use `global.require` for this dynamic import because of webpack.

If someone else need explanations on how to use this golden module in an angular-electron env you also have to : 

- Get rid of the require(...)(...) in the index.js of the "iconv-lite" module, __webpack__require() doesn't like it. Just comment this test (line 144 at this time)
`var nodeVerArr = nodeVer.split('.').map(Number);
   if (nodeVerArr[0] > 0 || nodeVerArr[1] >= 10) {
    require('iconv-lite');`
and replace `require('./extend-node')(iconv);` by `require('iconv-lite');`

- don't use the constant __dirname in index.js of this module because you'll get the absolute path of the electron renderer. I found this workaround:
`const path = require('path');
const { remote } = require('electron');
let appPath = remote.app.getAppPath();
module.exports = createApi(path.resolve(appPath, './node_modules/torrent-search-api/lib/providers'));
`

Again thanks a lot for this module and for your help.
